docs/user-guide/federation/federated-ingress.md

```diff
@@ -0,0 +1,323 @@
+
+Federated Ingress is released as beta for Google Cloud (GKE,
+GCE and hybrid scenarios involving both) in Kubernetes v1.4.  Work is under way to support other cloud
+providers (e.g. AWS) and other hybrid cloud scenarios (e.g. services
+spanning private on-premise as well as public cloud kubernetes
+clusters).  We welcome your feedback.  
+
+* TOC
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running Kubernetes Cluster
+Federation installation. If not, then head over to the
+[federation admin guide](/docs/admin/federation/) to learn how to
+bring up a cluster federation (or have your cluster administrator do
+this for you). Other tutorials, for example
+[this one](https://github.com/kelseyhightower/kubernetes-cluster-federation)
+by Kelsey Hightower, are also available to help you.
+
+You are also expected to have a basic
+[working knowledge of Kubernetes](/docs/getting-started-guides/) in
+general, and [Ingress](/docs/user-guide/ingress/) in particular.
+
+## Overview
+
+Federated Ingresses are created in much that same way as traditional
+[Kubernetes Ingresses](/docs/user-guide/ingress/) by making an API
+call which specifies the desired properties of your logical ingress point. In the
+case of Federated Ingress, this API call is directed to the
+Federation API endpoint, rather than a Kubernetes cluster API
+endpoint. The API for Federated Ingress is 100% compatible with the
+API for traditional Kubernetes Services.
+
+Once created, the Federated Ingress automatically:
+
+1. creates matching Kubernetes Ingress objects in every cluster
+underlying your Cluster Federation,
+2. ensures that all of these in-cluster ingress objects share the same
+   logical global L7 (i.e. HTTP(S)) load balancer and IP address.  
+3. monitors the health and capacity of the service "shards" (i.e. your
+   pods) behind this ingress in each cluster
+4. ensures that all client connections are routed to an appropriate
+healthy backend service endpoint at all times, even in the event of
+pod, cluster,
+availability zone or regional outages.
+
+Note that in the
+   case of Google Cloud, the logical L7 load balancer is not a single physical device (which
+   would present both a single point of failure, and a single global
+   network routing choke point), but rather a [truly global, highly available
+   load balancing managed service](https://cloud.google.com/load-balancing/),
+   reachable via a single, static IP address, globally.
+
+Clients inside your federated Kubernetes clusters (i.e. Pods) will be 
+automatically  routed to the cluster-local shard of the Federated Service
+backing the Ingress in their
+cluster if it exists and is healthy, or the closest healthy shard in a
+different cluster if it does not.  Note that this involves a network
+trip to the HTTP(s) load balancer, which resides outside your local
+Kubernetes cluster but inside the same GCP region.
+
+## Creating a federated ingress
+
+This is done in the usual way, for example:
```

Comment:
Disagree.  Kubectl is only one way of doing it.  It could also be achieved via any of the other "usual" ways, e.g. a GUI, or an API client.  Any suggestions on how best to word this?  Or perhaps we should leave as-is?  I'm open to suggestions.